### PR TITLE
docs: fix ssr warning

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/41563

### 💡 Background and solution
两种情况下的 ssr 问题，原因都可以归为一类：Layout 上的用户交互触发了 Suspense children 重新渲染

1. 页面载入时，点击左侧 menu
原因：切换路由导致整个在 Suspense 下的组件重新渲染，打断了注水过程
修复方法：做了一个 workaround，写了个新的 Link 组件，把路由跳转放到了 `startTransition` 里。但是这样做会有一个缺点，当前页面全部注水完毕才会触发跳转逻辑，所以整体上会有一点交互延迟（首屏场景）。
这个问题要根治需要框架层解决。

2. 页面载入时，点击右侧 Anchor
原因：react-router 通过 `hashchange` 监听 hash 变化，然后放在 context 里，通过 `useLocation` 在子组件中消费，导致重新渲染，打断注水。由于 a 标签在注水之前用户就可交互，暂时没什么好办法处理这种场景。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    -       |
| 🇨🇳 Chinese |      -     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8d7eaa1</samp>

Replaced `dumi`'s `Link` component with a custom one that uses concurrent mode and `startTransition` to improve page transitions. Added `onClick` handler to `Content` component to prevent browser navigation.

### <samp>🤖 Generated by Copilot at 8d7eaa1</samp>

* Replace `dumi`'s `Link` component with a custom one to enable concurrent mode and improve transition performance ([link](https://github.com/ant-design/ant-design/pull/41776/files?diff=unified&w=0#diff-97b7cacb6d4bd1445cc4d73ae072d3aac76d65d4c10eb29328c6cc8aef1d6649L4-R6), [link](https://github.com/ant-design/ant-design/pull/41776/files?diff=unified&w=0#diff-710e188e465a89d2e744c5183aa1ee6375dbefdcd24be26ad95e11ca7805e37bR1-R30))
* Prevent the default behavior of the anchor element in the `Content` component to avoid triggering the browser navigation ([link](https://github.com/ant-design/ant-design/pull/41776/files?diff=unified&w=0#diff-1533ad458dddb5cacbf24062cc5d262ca684a57e2754bf4777f24f37411d5f82R216))

### CODE REVIEW

File: .dumi/theme/slots/Content/index.tsx [hunk 1]
```diff
@@ -213,6 +213,7 @@ const Content: React.FC<{ children: ReactNode }> = ({ children }) => {
               affix={false}
               targetOffset={token.marginXXL}
               showInkInFixed
+              onClick={(e) => e.preventDefault()}
```
Comment (MadCcc) [inline .dumi/theme/slots/Content/index.tsx:216]: 似乎不能这样简单解决，注水完成之前点击还是会触发浏览器默认行为

File: .dumi/hooks/useLocation.ts [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
 import useLocale from './useLocale';
 
 function clearPath(path: string) {
   return path.replace('-cn', '').replace(/\/$/, '');
 }
 
 export default function useLocation() {
-  const location = useDumiLocation();
+  const originLocation = useDumiLocation();
+  const [location, setLocation] = React.useState(originLocation);
```
Comment (afc163) [inline .dumi/hooks/useLocation.ts:12]: 这样改的意义在哪里？感觉 originLocation 和 location 没啥区别？

File: .dumi/theme/slots/Content/index.tsx (hunk 1 above)
Comment (MadCcc) [reply .dumi/theme/slots/Content/index.tsx]: > 切这里也有问题。
> 
> 图片

这个应该 ok 了

**Comment by afc163** [comment]
>  这个问题要根治需要框架层解决。

框架层指的是 dumi、umi 还是 React？

**Review by li-jia-nan — APPROVED** [review]

**Comment by MadCcc** [comment]
> > 这个问题要根治需要框架层解决。
> 
> 框架层指的是 dumi、umi 还是 React？

提供 ssr 封装的 umi

**Pull request merged**